pkg/driver/node.go

```diff
@@ -226,5 +231,61 @@ func (d *nodeService) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 }
 
 func (d *nodeService) NodeGetVolumeStats(ctx context.Context, req *csi.NodeGetVolumeStatsRequest) (*csi.NodeGetVolumeStatsResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "NodeGetVolumeStats is not implemented yet")
+	klog.V(6).Infof("NodeGetVolumeStats: called with args %+v", req)
+
+	volumePath := req.GetVolumePath()
+	if len(volumePath) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume path not provided")
+	}
+
+	var corruptedMnt bool
+	var exists bool
+
+	err := util.DoWithTimeout(ctx, defaultCheckTimeout, func() (err error) {
+		exists, err = k8sMount.PathExists(volumePath)
+		return
+	})
+	if err == nil {
+		if !exists {
+			klog.V(5).Infof("NodeGetVolumeStats: %s Volume path not exists", volumePath)
+			return nil, status.Error(codes.Internal, "Volume path not exists")
+		}
+		var notMnt bool
+		err := util.DoWithTimeout(ctx, defaultCheckTimeout, func() (err error) {
+			notMnt, err = k8sMount.IsNotMountPoint(d.SafeFormatAndMount.Interface, volumePath)
+			return err
+		})
+		if err != nil {
+			klog.V(5).Infof("NodeGetVolumeStats: Check volume path %s is mountpoint failed: %s", volumePath, err)
+			return nil, status.Errorf(codes.Internal, "Check volume path is mountpoint failed: %s", err)
+		}
+		if notMnt { // target exists but not a mountpoint
+			klog.V(5).Infof("NodeGetVolumeStats: %s volume path not mounted", volumePath)
+			return nil, status.Error(codes.Internal, "Volume path not mounted")
+		}
+	} else if corruptedMnt = k8sMount.IsCorruptedMnt(err); !corruptedMnt {
```

Comment:
Why ignore corrupted errors?

Reply:
My mistake. Corrupted errors should be handled. Code has been updated.